libc/syscalls.c

```diff
@@ -235,23 +235,7 @@ long __syscall(long n, long x1, long x2, long x3, long x4, long x5, long x6)
         /* The hook ignored the syscall so fall through */
     }
 
-    /* Let liboesyscall handle select system calls. */
-    {
-        long ret;
-
-        errno = 0;
-
-        ret = _dispatch_oe_syscall(n, x1, x2, x3, x4, x5, x6);
-
-        if (!(ret == -1 && errno == ENOSYS))
-        {
-            return ret;
-        }
-
-        /* Drop through and let the code below handle the syscall. */
-        errno = 0;
-    }
-
+    /* Handle syscall internally if possible. */
     switch (n)
     {
         case SYS_nanosleep:
```

Comment:
Is there a reason these aren't included in _dispatch_oe_syscall() and separated by a switch here instead? (Not specifically directed at @jhand2 but a question for anyone!)

Reply:
The SYS_nanosleep case seems to make an OCALL, but the others seem to not. It makes reasonable sense to me to just bundle these cases all up with the _dispatch_oe_syscall() function, and that function is the one place in the code to refer to for how the enclave syscall handler behaves! What do you all think? :)

Reply:
I suppose the difference here is that all the syscalls in `_dispatch_oe_syscall` all eventually call `oe_syscall` and make an ocall.

You're right that `SYS_nanosleep` also makes an ocall, but it does it directly (with oe_ocall) rather than making it via `oe_syscall`. nanosleep is a little weird because it calls the `sleep` ocall, which is not actually the same as POSIX sleep (it takes in milliseconds rather than seconds), which in turn on linux calls nanosleep. `oe_sleep` just converts seconds to milliseconds and then calls this function. The Windows `Sleep` function also takes in milliseconds.

So one option for refactoring this code would be to have two syscalls handled by oe_syscall, sleep and nanosleep. On Linux they will call the correct respective syscalls. On Windows, they will both call Sleep, and convert to milliseconds when needed. I would leave `SYS_gettimeofday`, `SYS_clock_gettime`, `SYS_mmap` alone though since none of them make ocalls.

Reply:
I opened an issue here for this: #2405

I'd like to keep this PR targeted since it fixes a specific bug.